controllers/blogController.js

```diff
@@ -10,10 +10,12 @@ const blogs_get = async (req, res) => {
   }
 }
 
-const blogs_single_get = async (req, res) => {
+// Not working yet. Working on it.
+const blog_get = async (req, res) => {
+  const id = req.params.id;
     try {
       const blogs = await Blog.findById(id);
-      res.render("blog/blog.ejs", {blogs});
+      res.render("blog/blog/:id", {blogs, id});
```

Comment:
Ah, here is a problem. res.render() requires the name of the ejs file you're rendering, not the route. So, this will be: res.render("blog/blog.ejs", { blog }).